### DISCUSSION TITLE
Mix Panache and JDBC access?

### DISCUSSION THREAD

**Discussion by kenyee** [quarkusio/quarkus#25349] [Q&A]
Is it possible to mix Panache and JDBC access?
I have some Panache code that is taking unusually long to run when talking to CloudSQL....doesn't do anything complicated beyond adding a bunch of rows in two tables with one having foreign keys to the other so I can't see why it's taking so long.  Extending the transaction timeout to 5min still is timing out 🤔 

I was wondering if it'd be faster to do this using raw JDBC code...

[a comment by an automated account is not shown]

**Reply by loicmathieu** [reply]
You can always inject a JPA `EntityManager` and emit raw queries on it, or inject a `Datasource` and do what you want with it.
But there is no reason you observe such delay with Hibernate with Panache, there should be an issue somewhere.

**Comment by kenyee** [comment]
Thanks.  Didn't know that was possible.  Worth mentioning in the transaction docs (https://quarkus.io/guides/transaction or maybe https://quarkus.io/guides/hibernate-orm)?  The docs only mention using UserTransaction or injecting EntityManager...nothing about using EntityManager to do raw queries (though admittedly, you shouldn't have to).

Chased down my issue a bit...turned out the code was doing Cascade.ALL on the child objects as well as the parent.  This didn't manifest as a slowdown when testing against a local mysql container locally, but did in production because of the slower jdbc roundtrip times.
It's still a bit slow (adding 1000 root level rows takes 1.5min but it has a bunch of child rows w/ mapping tables and the generation of identity columns is DB side)...will see if doing raw SQL is faster...
